dusty/schemas/base_schema_class.py

```diff
@@ -19,8 +22,70 @@ def __delitem__(self, key):
     def __contains__(self, key):
         return key in self._document
 
+    def __iter__(self):
+        return iter(self._document)
+
+    def __len__(self):
+        return len(self._document)
+
     def keys(self):
         return self._document.keys()
 
     def values(self):
         return self._document.values()
+
+
+def _get_respective_schema(specs_type):
+    if specs_type == 'apps':
+        return app_schema
+    elif specs_type == 'bundles':
+        return bundle_schema
+    elif specs_type == 'libs':
+        return lib_schema
+    elif specs_type == 'services':
+        return None
+    else:
+        raise RuntimeError('Specs must be of the type apps, bundles, libs or services')
+
+
+# This is build on top of Schemer's functionality
+class DustySchema(BaseMutable):
+    def __init__(self, schema, document, name, spec_type):
+        if schema is not None:
+            schema.validate(document)
+        self.name = name
+        self.spec_type = spec_type
+        self._document = deepcopy(document)
+        super(DustySchema, self).__init__(deepcopy(document))
+        if schema is not None:
+            schema.apply_defaults(self._document)
+
+
+class DustySpecs(BaseMutable):
+    def __init__(self, specs_path):
+        document = self.get_specs_from_path(specs_path)
+        super(DustySpecs, self).__init__(document)
+
+    def get_specs_from_path(self, specs_path):
+        specs = {}
+        for key in ['bundles', 'apps', 'libs', 'services']:
+            specs[key] = {}
+            schema = _get_respective_schema(key)
+            key_path = os.path.join(specs_path, key)
+            for spec_path in glob.glob('{}/*.yml'.format(key_path)):
+                spec_name = os.path.splitext(os.path.split(spec_path)[-1])[0]
+                with open(spec_path, 'r') as f:
+                    spec = yaml.load(f.read())
```

Comment:
As in a lib can't have the same name as an app?  I thought we just needed uniqueness within a given type, and each spec of a given type lives in the same dir